internals/src/lib.rs

```diff
@@ -35,3 +35,39 @@ mod parse;
 #[cfg(feature = "serde")]
 #[macro_use]
 pub mod serde;
+
+/// Reads a `usize` from an iterator of bytes.
+// TODO: Where should this live?
+pub fn read_uint_iter(
```

Comment:
What's this about? Why it can't be in `bitcoin`?

Reply:
It's used in `primitives`, in this PR.